sympy/combinatorics/fp_groups.py

```diff
@@ -279,6 +285,54 @@ def __str__(self):
     __repr__ = __str__
 
 
+class FpSubgroup(DefaultPrinting):
+    '''
+    The class implementing a subgroup of an FpGroup or a FreeGroup
+    (only finite index subgroups are supported at this point). This
+    is to be used if one wishes to check if an element of the original
+    group belongs to the subgroup
+
+    '''
+    def __init__(self, G, gens):
+        super(FpSubgroup,self).__init__()
+        self.parent = G
+        self.generators = list(set([g for g in gens if g != G.identity]))
+        self.C = None
+
+    def __contains__(self, g):
+        gens = [s for s in g.contains_generators()]
+        if any([s not in self.generators for s in gens]):
```

Comment:
This looks like a strong condition. Is it necessary that each one of the generators appearing in `g` should be among the subgroup generators?